src/lib.rs

```diff
@@ -215,14 +229,14 @@ pub struct TestFile {
 }
 
 impl TestFile {
-    fn new(filename: &str) -> Result<Self, Error> {
-        let contents = fs::read_to_string(filename)
-            .with_context(|| format!("error reading file {}", filename))?;
+    fn new(filename: &PathBuf) -> Result<Self, Error> {
+        let contents = fs::read_to_string(filename.to_str().unwrap())
```

Comment:
Just `filename` here works! `read_to_string` takes an `AsRef<Path>`, so converting to a str is unnecessary work.